rubix/core/data.py

```diff
@@ -10,6 +12,118 @@
 from rubix.galaxy.alignment import center_particles
 from rubix.logger import get_logger
 from rubix.utils import load_galaxy_data, read_yaml
+from rubix import config as rubix_config
+
+
+# Registering the dataclass with JAX for automatic tree traversal
+@partial(jax.tree_util.register_pytree_node_class)
+@dataclass
+class Galaxy:
+    redshift: Optional[jnp.ndarray] = None
+    center: Optional[jnp.ndarray] = None
+    halfmassrad_stars: Optional[jnp.ndarray] = None
+
+    def tree_flatten(self):
+        children = (self.redshift, self.center, self.halfmassrad_stars)
+        aux_data = {}
+        return children, aux_data
+
+    @classmethod
+    def tree_unflatten(cls, aux_data, children):
+        return cls(*children)
+
+@partial(jax.tree_util.register_pytree_node_class)
```

Comment:
we should have a Look if we want to use [equinox module](https://docs.kidger.site/equinox/api/module/module/) or [chex dataclass](https://chex.readthedocs.io/en/latest/api.html#chex.dataclass) and what the differences are

Reply:
I think this automatically impelements the class as a pytree

Reply:
@ufuk-cakir the same discussion is ongoing on the branch final load gas particles

Reply:
equinox has a problem, if some values in the beginning are None and then set to a jax.array

Reply:
We can change the initialisation of the class. @ufuk-cakir could you please provide a suggestion how to change the code?

Reply:
what about instead of dataclass we use python native `NamedTuple` ? It looks like there is not really any functionality defined for the dataclass, but rather used to define which variables the data structure holds. As far as I know, JAX works with NamedTuples, and gradient computation should also work.

Reply:
I think the notebook branch is the wrong branch to do large changes on the dataclass of the rubixdata. If you want to change it, please open a new issue and branch, otherwise it gets a too large mess on the individual branches and it is difficult to understand the changes

Reply:
If we have the spectra or datacube as field in the rubixdata, this is when initializing the dataclass not peresent at that point, because it is calculated later in the pipeline

Reply:
> what about instead of dataclass we use python native `NamedTuple` ? It looks like there is not really any functionality defined for the dataclass, but rather used to define which variables the data structure holds. As far as I know, JAX works with NamedTuples, and gradient computation should also work.

also moved this to [issue](https://github.com/ufuk-cakir/rubix/issues/119) so we have a reference.
Can we merge, so we can progress with all other PRs?